In [137]:
import pandas as pd

### Data Preparation and Cleaning

The dataset is combined from TMDB API and OMDB API.

You can look at the scraping process in scrape_data.ipynb

In [138]:
# The dataset is combined from TMDB API and OMDB API.
# You can look at the scraping process in scrape_data.ipynb
df = pd.read_parquet('../data/movies.parquet')

Let's take a look at the shape, features that are available to us and preview the data

Dataset is X movies with 58 metadata columns/features. 

In [139]:
print(df.shape)
print(df.columns)
df.head(10)

(5760, 59)
Index(['adult', 'backdrop_path', 'belongs_to_collection', 'budget', 'genres',
       'homepage', 'id', 'imdb_id', 'original_language', 'original_title',
       'overview', 'popularity', 'poster_path', 'production_companies',
       'production_countries', 'release_date', 'revenue', 'runtime',
       'spoken_languages', 'status', 'tagline', 'title', 'video',
       'vote_average', 'vote_count', 'Title', 'Year', 'Rated', 'Released',
       'Runtime', 'Genre', 'Director', 'Writer', 'Actors', 'Plot', 'Language',
       'Country', 'Awards', 'Poster', 'Ratings', 'Metascore', 'imdbRating',
       'imdbVotes', 'imdbID', 'Type', 'totalSeasons', 'Response', 'DVD',
       'BoxOffice', 'Production', 'Website', 'belongs_to_collection.id',
       'belongs_to_collection.name', 'belongs_to_collection.poster_path',
       'belongs_to_collection.backdrop_path', 'Season', 'Episode', 'seriesID',
       'Error'],
      dtype='object')


adult                     backdrop_path  belongs_to_collection    budget  \
0  False                              None                    NaN         0   
1  False  /7u6P46nr6vNua00DmKOyzfrqTv1.jpg                    NaN         0   
2  False  /5r2BZajlRZqnOc6s2BS0aiFDcne.jpg                    NaN  30000000   
3  False  /w8VDpCcKezdBbXJEWMHwttI35mU.jpg                    NaN  30000000   
4  False  /jWZqmct37aYQKB6TU9IaTg6346G.jpg                    NaN  35000000   
5  False  /3CEKwdKVLfSCzeAOpqVLquLwGQQ.jpg                    NaN         0   
6  False  /7NvMR6ihwuH8aXQ8PnVxSpxaMxV.jpg                    NaN  55000000   
7  False  /cCzvJ95pEB1I5cUX7Zjfl606noN.jpg                    NaN  30000000   
8  False  /ii3WNOfgEp62jkJAHAOJyYwRfhl.jpg                    NaN         0   
9  False  /7Pc719slk1WDC4fH1pk28pdPemv.jpg                    NaN         0   

                                              genres  \
0                      [{'id': 18, 'name': 'Drama'}]   
1  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...   
2  [{'id': 28, 'name': 'Action'}, {'id': 18, 'nam...   
3  [{'id': 27, 'name': 'Horror'}, {'id': 53, 'nam...   
4  [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...   
5                     [{'id': 35, 'name': 'Comedy'}]   
6  [{'id': 18, 'name': 'Drama'}, {'id': 36, 'name...   
7  [{'id': 18, 'name': 'Drama'}, {'id': 36, 'name...   
8  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...   
9  [{'id': 18, 'name': 'Drama'}, {'id': 9648, 'na...   

                                 homepage    id    imdb_id original_language  \
0                                          4501  tt0335663                de   
1                                          4507  tt0104545                es   
2  http://jessejamesmovie.warnerbros.com/  4512  tt0443680                en   
3           http://www.30daysofnight.com/  4513  tt0389722                en   
4                                          4515  tt0891527                en   
5   http://www.monmeilleurami-lefilm.com/  4516  tt0778784                fr   
6                                          4517  tt0414055                en   
7                                          4518  tt0127536                en   
8                                          4519  tt0825244                fr   
9     http://www.sonyclassics.com/sleuth/  4520  tt0857265                en   

                                      original_title  ... Production  Website  \
0                                  Der Fall Rohrbach  ...       None     None   
1                                       Jamón, jamón  ...        N/A      N/A   
2  The Assassination of Jesse James by the Coward...  ...        N/A      N/A   
3                                   30 Days of Night  ...        N/A      N/A   
4                                    Lions for Lambs  ...        N/A      N/A   
5                                   Mon Meilleur Ami  ...        N/A      N/A   
6                          Elizabeth: The Golden Age  ...        N/A      N/A   
7                                          Elizabeth  ...        N/A      N/A   
8                        Dialogue avec mon jardinier  ...        N/A      N/A   
9                                             Sleuth  ...        N/A      N/A   

  belongs_to_collection.id   belongs_to_collection.name  \
0                      NaN                         None   
1                      NaN                         None   
2                      NaN                         None   
3                  91660.0  30 Days of Night Collection   
4                      NaN                         None   
5                      NaN                         None   
6                 100286.0         Elizabeth Collection   
7                 100286.0         Elizabeth Collection   
8                      NaN                         None   
9                      NaN                         None   

  belongs_to_collection.poster_path belongs_to_collection.backdrop_path  \
0                 

In [140]:
df.describe()

belongs_to_collection        budget            id   popularity  \
count                    0.0  5.760000e+03   5760.000000  5760.000000   
mean                     NaN  1.488827e+07   5399.694271    14.410539   
std                      NaN  2.967762e+07   3533.584911    16.295273   
min                      NaN  0.000000e+00      2.000000     0.000000   
25%                      NaN  0.000000e+00   2101.750000     3.641000   
50%                      NaN  0.000000e+00   5066.500000    11.372000   
75%                      NaN  1.760000e+07   9091.250000    18.683750   
max                      NaN  3.790000e+08  10832.000000   302.315000   

            revenue      runtime  vote_average    vote_count  \
count  5.760000e+03  5760.000000   5760.000000   5760.000000   
mean   4.516967e+07   103.435764      6.046570   1173.774826   
std    1.130256e+08    26.307339      1.726494   2591.248658   
min    0.000000e+00     0.000000      0.000000      0.000000   
25%    0.000000e+00    90.000000      5.665750     23.000000   
50%    0.000000e+00   100.000000      6.400000    241.000000   
75%    3.676167e+07   114.000000      7.030000   1090.750000   
max    2.264162e+09   460.000000     10.000000  30664.000000   

       belongs_to_collection.id  
count              1.176000e+03  
mean               2.163804e+05  
std                2.942376e+05  
min                1.000000e+01  
25%                9.436000e+03  
50%                9.745950e+04  
75%                2.615900e+05  
max                1.177319e+06

In [141]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 5760 entries, 0 to 6009
Data columns (total 59 columns):
 #   Column                               Non-Null Count  Dtype  
---  ------                               --------------  -----  
 0   adult                                5760 non-null   bool   
 1   backdrop_path                        5004 non-null   object 
 2   belongs_to_collection                0 non-null      float64
 3   budget                               5760 non-null   int64  
 4   genres                               5760 non-null   object 
 5   homepage                             5760 non-null   object 
 6   id                                   5760 non-null   int64  
 7   imdb_id                              5760 non-null   object 
 8   original_language                    5760 non-null   object 
 9   original_title                       5760 non-null   object 
 10  overview                             5760 non-null   object 
 11  popularity                         

Let's have a look at columns and select only relevant ones

In [142]:
# title vs original title: so we can see that original_title is in the native language, so we can drop it and leave only title
df[df['original_title'] != df['title']][['title', 'original_title']].head(10)

title  \
1                       Jamon Jamon   
5                    My Best Friend   
8    Conversations with My Gardener   
11  Death Is Nimble, Death Is Quick   
12             Kiss Kiss, Kill Kill   
13        The Count of Monte Cristo   
14                     The Defector   
15            So Darling, So Deadly   
16                   Kill Me Gently   
17             Kill, Panther, Kill!   

                                      original_title  
1                                       Jamón, jamón  
5                                   Mon Meilleur Ami  
8                        Dialogue avec mon jardinier  
11                   Kommissar X - Drei gelbe Katzen  
12                  Kommissar X - Jagd auf Unbekannt  
13                          Le Comte de Monte-Cristo  
14                                          L'espion  
15  Kommissar X - In den Klauen des goldenen Drachen  
16                    Kommissar X - Drei grüne Hunde  
17                  Kommissar X - Drei blaue Panther

In [143]:
# status of the movie, we will actually need only released movie so the data should be filtered
print(df['status'].value_counts())
df = df[df['status'].str.lower() == 'released']

status
Released           5756
Post Production       2
Planned               1
Canceled              1
Name: count, dtype: int64


In [144]:
# We have too many False in adult columns and mostly NULL values in belongs_to_collection.
# Those 2 features are not of much use to us
print(df['adult'].value_counts())
print(df['belongs_to_collection'].value_counts())

adult
False    5744
True       12
Name: count, dtype: int64
Series([], Name: count, dtype: int64)


In [145]:
df.head().transpose()

0  \
adult                                                                            False   
backdrop_path                                                                     None   
belongs_to_collection                                                              NaN   
budget                                                                               0   
genres                                                   [{'id': 18, 'name': 'Drama'}]   
homepage                                                                                 
id                                                                                4501   
imdb_id                                                                      tt0335663   
original_language                                                                   de   
original_title                                                       Der Fall Rohrbach   
overview                                                                                 
popularity                                                                       1.093   
poster_path                                                                       None   
production_companies                                                                []   
production_countries                         [{'iso_3166_1': 'DE', 'name': 'Germany'}]   
release_date                                                                1963-05-08   
revenue                                                                              0   
runtime                                                                            270   
spoken_languages                     [{'english_name': 'German', 'iso_639_1': 'de',...   
status                                                                        Released   
tagline                                                                                  
title                                                                Der Fall Rohrbach   
video                                                                            False   
vote_average                                                                       0.0   
vote_count                                                                           0   
Title                                                                Der Fall Rohrbach   
Year                                                                             1963–   
Rated                                                                              N/A   
Released                                                                   08 May 1963   
Runtime                                                                        270 min   
Genre                                                                              N/A   
Director                                                                           N/A   
Writer                                                                             N/A   
Actors                                      Anita Höfer, Eva Christian, Harry Riebauer   
Plot                                                                               N/A   
Language                                                                        German   
Country                                                                   West Germany   
Awards                                                                             N/A   
Poster                                                                             N/A   
Ratings                              [{'Source': 'Internet Movie Database', 'Value'...   
Metascore                                                                          N/A   
imdbRating                                                                         9.1   
imdbVotes                                                                           15   
imdbID                                                                       tt0335663   
Type                               

In [146]:
# we have release_date column which have day, month and year so other date related columns as Year and Released can be dropped
df[df['release_date'].isnull()].shape #- zero NULL values

(0, 59)

In [147]:
# let's look at how many NULLs other fields contain
var = df.isnull().sum()
print(var.to_string())

adult                                     0
backdrop_path                           752
belongs_to_collection                  5756
budget                                    0
genres                                    0
homepage                                  0
id                                        0
imdb_id                                   0
original_language                         0
original_title                            0
overview                                  0
popularity                                0
poster_path                             228
production_companies                      0
production_countries                      0
release_date                              0
revenue                                   0
runtime                                   0
spoken_languages                          0
status                                    0
tagline                                   0
title                                     0
video                           

In [148]:
# we will need only movies, because others are incompariable categories
print(df['Type'].value_counts())
df[df['Type'].str.lower() == 'series'][['title', 'original_title']]
df = df[df['Type'].str.lower() == 'movie']

Type
movie      5725
series       19
episode      12
Name: count, dtype: int64


Let's summaries our findings on columns

| Column | Meaning | Take as a feature? | Why? | 
| :- | :- | :- | :- |
| adult | if the movie is X-Rated or Adult | No | too many identical values (False), not usefull |
| backdrop_path | path to image | No | not usefull |
| belongs_to_collection | the movie series the particular film belongs to | No | almost all NULL |
| genres | the genres associated with the movie | Yes | - |
| homepage | path to the movie homepage | No | not usefull |
| id | TMDB id | No | not usefull |
| imdb_id | IMDB id | No | not usefull |
| original_language | the language in which the movie was originally shot in | Yes | - |
| original_title | The original title of the movie in the native language | No | not usefull |
| overview | short description of the movie | Yes | - |
| poster_path | path to image | No | not usefull |
| production_companies | list of production companies involved with the making of the movie | Yes | - |
| production_countries | list of countries where the movie was shot/produced in | Yes | - |
| release_date |  | Yes | - |
| runtime | the runtime of the movie in minutes | Yes | - |
| spoken_languages | list of spoken languages in the film | Yes | - |
| status | status of the movie | No | we take only Released status |
| tagline | tagline of the movie | Yes | - |
| status | status of the movie | No | we take only Released status |
| title | title of the movie | Yes | - |
| video | Indicates if there is a video present of the movie with TMDB | No | not usefull |
| vote_count | number of votes by users, as counted by TMDB | Yes | - |
| Title |  | No | duplicate of title |
| Year |  | No | duplicate of year in the release_date |
| Rated | MPA film rating | Yes | - |
| Released |  | No | duplicate of release_date |
| Runtime |  | No | duplicate of runtime |
| Genre |  | No | duplicate of genres |
| Plot |  | Yes | - |
| Language |  | No | duplicate of original_language |
| Country |  | No | duplicate of production_countries |
| Poster | path to poster image | No | not usefull |
| imdbVotes | count of voutes | Yes | - |
| imdbID | IMDB id | No | not usefull |
| Type | status of the movie | No | we filtered only movie type |
| totalSeasons |  | No | Not relevant |
| Season |  | No | Not relevant |
| Episode |  | No | Not relevant |
| seriesID |  | No | Not relevant |
| Error |  | No | Not relevant |
| belongs_to_collection.id|  | No | Almost all rows are NULL |
| belongs_to_collection.name|  | No | Almost all rows are NULL |
| belongs_to_collection.poster_path|  | No | Almost all rows are NULL |
| belongs_to_collection.backdrop_path|  | No | Almost all rows are NULL |
| DVD | the date when the movie was released on dvd | No | Not useful |
| Website | link | No | Not useful |
| BoxOffice |  | No | We have revenue column plus too many NULLS |
| Production |  | No | We have production_companies column |
| Response | link | No | Not relevant |
| Ratings | list of ratings in different system - metacritic (in Metascore), rotten tomatoes, imdb (in imdbRating) | Yes | |
| Metascore | metacritic rating | Yes | - |
| imdbRating | imdb raiting | Yes | - |
| vote_average | average rating of the movie by TMDB | Yes | - |
| popularity | popularity Score assigned by TMDB | Yes | - |
| budget | the budget of the movie in dollars | Yes | - |
| revenue | the total revenue of the movie in dollars | Yes | - |
| Awards |  | Yes | - |


In [174]:
# 'Director', 'Writer', 'Actors',
dataset_df = df[['title', 'release_date', "original_language" , 'spoken_languages', 'genres',
                       'production_companies',  'production_countries', 'runtime', 'Rated', 'tagline', 'overview', 'Plot',
                       'budget', 'revenue', 'popularity', 'vote_average', 'vote_count',
                         'Awards', 'Ratings', 'Metascore', 'imdbRating', 'imdbVotes']].copy()

# Standardize column names
dataset_df.columns = dataset_df.columns.str.lower().str.replace(' ', '_')
dataset_df.rename(columns={"popularity": "tmdb_popularity", "vote_average": "tmdb_vote_average",
                     "vote_count": "tmdb_vote_count", 'metascore': 'metacritic_rating',
                     "imdbrating": "imdb_rating", 'imdbvotes': 'imdb_votes'}, inplace=True)

# let's unpack values from the dictionaries to lists for columns: genres, production_companies, production_countries, spoken_languages
# same for spoken_languages and production_countries
dataset_df['genres'] = dataset_df['genres'].apply(lambda x: ','.join([d['name'] for d in x]))
dataset_df['production_companies'] = dataset_df['production_companies'].apply(lambda x: ','.join([d['name'] for d in x]))
dataset_df['production_countries'] = dataset_df['production_countries'].apply(lambda x: ','.join([d['iso_3166_1'] for d in x]))
dataset_df['spoken_languages'] = dataset_df['spoken_languages'].apply(lambda x: ','.join([d['iso_639_1'] for d in x]))

#let's extract rotten tomatoes rating from Ratings column
# and convert other ratings to numeric values

def extract_rating(ratings, source):
    for rating in ratings:
        if rating['Source'] == source:
            return rating['Value']
    return None

dataset_df['rotten_tomatoes_rating'] = dataset_df['ratings'].apply(lambda x: extract_rating(x, 'Rotten Tomatoes')) 
dataset_df['rotten_tomatoes_rating'] = dataset_df['rotten_tomatoes_rating'].str.replace('%', '')

dataset_df['imdb_rating'] = pd.to_numeric(dataset_df['imdb_rating'], errors='coerce')
dataset_df['imdb_votes'] = dataset_df['imdb_votes'].str.replace(',', '')
dataset_df['imdb_votes'] = pd.to_numeric(dataset_df['imdb_votes'], errors='coerce')
dataset_df['rotten_tomatoes_rating'] = pd.to_numeric(dataset_df['rotten_tomatoes_rating'], errors='coerce')
dataset_df['metacritic_rating'] = pd.to_numeric(dataset_df['metacritic_rating'], errors='coerce')

# get relase year and release month
dataset_df['release_month'] = pd.to_datetime(dataset_df['release_date']).dt.month
dataset_df['release_year'] = pd.to_datetime(dataset_df['release_date']).dt.year
dataset_df.drop(['release_date', 'ratings'], axis=1, inplace=True)

# lower all values in categorical columns
categorical_vars = list(dataset_df.dtypes[dataset_df.dtypes == 'object'].index)
for c in categorical_vars:
    dataset_df[c] = dataset_df[c].str.lower()

pd.set_option('display.max_columns', None)
dataset_df.head(2)


title original_language  \
1                                        jamon jamon                es   
2  the assassination of jesse james by the coward...                en   

  spoken_languages                genres  \
1               es  comedy,drama,romance   
2               en  action,drama,western   

                                production_companies production_countries  \
1                   ovídeo tv s.a.,sogepaq,lolafilms                   es   
2  plan b entertainment,virtual studios,scott fre...             ca,gb,us   

   runtime rated                                           tagline  \
1       95     r        a film where women eat men and men eat ham   
2      160     r  beyond the myth lies america's greatest betrayal   

                                            overview  \
1  jose luis is an executive at his parents under...   
2  outlaw jesse james is rumored to be the 'faste...   

                                                plot    budget   revenue  \
1  when a mother disapproves of her son's lover, ...         0         0   
2  robert ford, who's idolized jesse james since ...  30000000  14711793   

   tmdb_popularity  tmdb_vote_average  tmdb_vote_count  \
1           16.879              6.221              306   
2           22.557              7.049             2276   

                                              awards  metacritic_rating  \
1                            6 wins & 12 nominations                NaN   
2  nominated for 2 oscars. 25 wins & 69 nominatio...               68.0   

   imdb_rating  imdb_votes  rotten_tomatoes_rating  release_month  \
1          6.4     14634.0                    69.0              9   
2          7.5    189145.0                    77.0              9   

   release_year  
1          1992  
2          2007

In [165]:
# let's look at awards column
dataset_df[['awards']].to_csv("1.csv")


So you might notice the following patterns:

    - n/a
    - X win/wins
    - Y nomination/nominations
    - X win/wins & Y nomination/nominations
    - nominated for A oscar/oscars. X win/wins & Y nomination/nominations total
    - won A oscar/oscars. 1X win/wins & Y nomination/nominations total
    - nominated for B bafta X win/wins & Y nomination/nominations total'
    - won B bafta X win/wins & Y nomination/nominations total
    - won E primetime emmy. X win/wins & Y nomination/nominations total
    - nominated for E primetime emmys

Let's use regex and unpack those values to the separate columns but before I noticed "emmy" award so we should check genres and filter the dataset on those which are different categories of movies in terms of budget or awards.  

In [179]:
# need to check genres
import pandas as pd
from collections import Counter
import ast  # if needed to convert string representation of list to list

genres_df = dataset_df.copy()
genres_df['genres'] = genres_df['genres'].str.split(',')
# Flatten the list of lists and count the occurrences of each country code
genre_counter = Counter(genre for genre_list in genres_df['genres'].dropna() for genre in genre_list)
# Convert the Counter object to a DataFrame for easier viewing and analysis
genre_frequencies = pd.DataFrame(genre_counter.items(), columns=['genres', 'frequency'])

# Sort the DataFrame by frequency
genre_frequencies = genre_frequencies.sort_values(by='frequency', ascending=False).reset_index(drop=True)
genre_frequencies

genres  frequency
0             drama       2763
1            comedy       1982
2          thriller       1286
3            action       1029
4             crime        934
5           romance        920
6         adventure        714
7            horror        621
8   science fiction        575
9           fantasy        417
10          mystery        406
11           family        318
12          history        258
13              war        191
14      documentary        172
15        animation        163
16            music        149
17         tv movie        122
18          western        111
19                          18

In [192]:
print(dataset_df.shape)

(5725, 23)


So we have 172 movie with Documentary, 122 where tv movie is mentioned and 18 movies without genre. Let's look at them and count total amount of row

In [184]:
# Filter rows where 'tv movie' or 'documentary' is in the 'Genres' column
outliers = dataset_df[dataset_df['genres'].apply(lambda x: 'tv movie' in x or 'documentary' in x or not x)][['title', 'awards', 'genres']]
print(outliers.shape)
outliers

(312, 3)


title  \
19                       the world's biggest penis   
22    hearts of darkness: a filmmaker's apocalypse   
28                       the count of monte-cristo   
80                                read it and weep   
90                                     arctic tale   
138                                   here we come   
143                              heart of darkness   
195    dungeons & dragons: wrath of the dragon god   
212                                    microcosmos   
269                                  super size me   
352                                      woodstock   
360                               margarete steiff   
398            the basque ball: skin against stone   
414                        der lange weg ans licht   
419                              he's not your son   
442                              mail to the chief   
444                  der butler und die prinzessin   
456                            dancing with danger   
467                                   mord am meer   
468         gegenschuss - aufbruch der filmemacher   
509                            cry of the innocent   
527                                 miracle on ice   
528            for those who hunt the wounded down   
534              samaritan: the mitch snyder story   
537            walker, texas ranger: trial by fire   
559                                    horse sense   
561                             the crooked hearts   
628                            nanook of the north   
636      olympia part one: festival of the nations   
642           olympia part two: festival of beauty   
...                                            ...   
5560                 freiwild - ein würzburg-krimi   
5584                  standard operating procedure   
5595                   someday my prince will come   
5612                             waltz with bashir   
5620                it's hard being loved by jerks   
5634                         maradona by kusturica   
5636                                 silberdisteln   
5647                                 i want to see   
5649                                   modern life   
5658                                         tyson   
5669                    white man with black bread   
5683                  the society of the spectacle   
5690                             visions of europe   
5707                              biùtiful cauntri   
5709                  küss mich, wenn es liebe ist   
5712                                        carrie   
5729               frühstück mit einer unbekannten   
5753                              queens don’t cry   
5754                                   pfui, rosa!   
5755                        cows knocked up by fog   
5792                                a hero's death   
5799                                  shadow realm   
5861                            destination saturn   
5871                     el camino de las hormigas   
5940                               railway station   
5941                             short working day   
5945                                 gimme shelter   
5946                                       primary   
5948                    bob dylan - dont look back   
5972           the arrival of a train at la ciotat   

                                                 awards  \
19                                                  n/a   
22    won 2 primetime emmys. 8 wins & 5 nominations ...   
28    nominated for 2 primetime emmys. 2 nominations...   
80                                        2 nominations   
90                                 1 win & 1 nomination   
138                                        1 nomination   
143   won 1 primetime emmy. 3 wins & 3 nominations t...   
195                                                 n/a   
212                              9 wins & 9 nominations   
269   nominated for 1 oscar. 7 wins & 13 nominations...   
352           won 1 oscar. 2 wins & 2 nominatio

Let's filter those rows

In [194]:
dataset_df = dataset_df[dataset_df['genres'].apply(lambda x: 'tv movie' not in x and 'documentary' not in x )]
dataset_df = dataset_df[dataset_df['genres'].apply(lambda x: len(x) > 0)]
print(dataset_df.shape)

(5413, 23)


Now we can use regex on awards column.

In [195]:
import re


pattern = {'oscar_won': re.compile(r'won (\d+) oscar', re.IGNORECASE),
           'oscar_nominated': re.compile(r'nominated for (\d+) oscar', re.IGNORECASE),
           'bafta_won': re.compile(r'won (\d+) bafta', re.IGNORECASE),
           'bafta_nominated': re.compile(r'nominated for (\d+) bafta', re.IGNORECASE),
           'awards_won': re.compile(r'(\d+)\s*win', re.IGNORECASE),
           'awards_nominated': re.compile(r'(\d+)\s*nomination', re.IGNORECASE)}
dataset_df['awards'] = dataset_df['awards'].str.replace('n/a', '')
for k, v in pattern.items():
# Use str.extract() to extract the number of Oscars won
    dataset_df[k] = dataset_df['awards'].str.extract(v).fillna(0).astype(int)



In [111]:


# Actors, Writers - I guess we should replace them with info from credits dataset - TODO

### Exploratory data analysis (EDA)

#### 1. Analysis of features / Feature Engineering

We're going to predict movie success so the definition of success according to the dataset can be defined though 3 metrics:
* revenue
* awards
* popularity/rating

So every piece of metadata which is not related to multidimensional SUCCESS - revenue, awards and popularity/rating will be our features. Let's look at them and get rid of obvious dublications that appeared because we merged 2 datasets

#### 2. Analysis of target variables